maybe-uninitialized warning with optimized Bazel build

Steps to reproduce (on Debian Buster-like system with GCC 7.3.0):

```
git clone https://github.com/jupp0r/prometheus-cpp
cd prometheus-cpp
bazel build -c opt //...
```

This produces the warning:

```
INFO: From Compiling lib/summary.cc:
In file included from lib/summary.cc:1:0:
bazel-out/k8-opt/bin/_virtual_includes/prometheus_cpp/prometheus/summary.h: In constructor 'prometheus::detail::TimeWindowQuantiles::TimeWindowQuantiles(const std::vector<prometheus::detail::CKMSQuantiles::Quantile>&, std::chrono::_V2::steady_clock::duration, int)':
bazel-out/k8-opt/bin/_virtual_includes/prometheus_cpp/prometheus/summary.h:17:7: warning: '<anonymous>.prometheus::detail::CKMSQuantiles::buffer_' may be used uninitialized in this function [-Wmaybe-uninitialized]
 class CKMSQuantiles {
       ^~~~~~~~~~~~~
```

The warning seems to only occur without `-fPIC`. The default `fastbuild` configuration always builds with `-fPIC`, but adding `-c opt` causes each object to be built twice, with and without `-fPIC`, resulting in this warning. I searched briefly but didn't find anywhere discussing an interaction between `-fPIC` and `-Wmaybe-uninitialized`.

Do you think the warning is meaningful? If not, should we suppress it by adding `"-Wno-maybe-uninitialized"` to `copts`?

--- a/lib/summary.cc
+++ b/lib/summary.cc
@@ -1,7 +1,7 @@
 #include "prometheus/summary.h"
 
-#include <cmath>
 #include <algorithm>
+#include <cmath>
 
 namespace prometheus {
 
@@ -17,7 +17,7 @@ CKMSQuantiles::Item::Item(double value, int lower_delta, int delta)
     : value(value), g(lower_delta), delta(delta) {}
 
 CKMSQuantiles::CKMSQuantiles(const std::vector<Quantile>& quantiles)
-    : quantiles_(quantiles), count_(0), buffer_count_(0) {}
+    : quantiles_(quantiles), count_(0), buffer_{}, buffer_count_(0) {}
 
 void CKMSQuantiles::insert(double value) {
   buffer_[buffer_count_] = value;

